include/ignition/gazebo/detail/EntityComponentManager.hh

```diff
@@ -20,15 +20,61 @@
 #include <cstring>
 #include <map>
 #include <set>
+#include <type_traits>
 #include <utility>
 #include <vector>
 
+#include <ignition/math/Helpers.hh>
+
 #include "ignition/gazebo/EntityComponentManager.hh"
 
 namespace ignition
 {
 namespace gazebo
 {
+//////////////////////////////////////////////////
+namespace traits
+{
+  /// \brief Helper struct to determine if an equality operator is present.
+  struct TestEqualityOperator
+  {
+  };
+  template<typename T>
+  TestEqualityOperator operator == (const T&, const T&);
+
+  /// \brief Type trait that determines if an operator== is defined for `T`.
+  template<typename T>
+  struct HasEqualityOperator
+  {
+    enum
+    {
+      value = !std::is_same<decltype(*(T*)(0) == *(T*)(0)),
```

Comment:
Codecheck doesn't like this because it is a C-style cast.  I don't see an obvious way around it, so I think a `// NOLINT` would be sufficient.